design/one-pager-stacks-security-isolation.md

```diff
@@ -109,57 +139,56 @@ The diagram below summarizes the artifacts in the control plane after the GCP st
 
 ![Diagram of a Cluster Stack Installed](./images/cluster-stack-installed.png)
 
-### Environments
+### Environment
 
-While cluster stacks are installed at the cluster level (there is only one instance of each cluster stack), they can still support the concept of multiple "environments", each represented by a namespace.
-For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
-All resources within each of those environments can share a common set of provider credentials.
-This namespaced "environment" approach provides the administrator with a way to collect or aggregate all the infrastructure and services associated with a given environment.
+Cluster stacks are installed at the cluster level (there is only one instance of each cluster stack), and there will
+ be a single environment for infrastructure which will be represented by the crossplane-system namespace as well as
```

Comment:
> will be represented by the crossplane-system namespace as well as cluster scoped resources

There's nothing special about `crossplane-system` as far as any Crossplane logic is concerned; I don't think we can assume it's a special infrastructure namespace. Today:

* Providers, resource classes, and managed resources are cluster scoped.
* Provider and managed resource secrets are written _by convention_ to `crossplane-system`.
* The use of `crossplane-system` as a place to read and write secrets is configurable by the infrastructure admin.

```yaml
apiVersion: gcp.crossplane.io/v1alpha3
kind: Provider
metadata:
  name: example
spec:
  credentialsSecretRef:
    # If you change this the provider will read its secret from the namespace
    # you specify. This is a required field; there is no default.
    namespace: crossplane-system
    name: example-provider-gcp
    key: credentials.json
  projectID: PROJECT_ID
```

```yaml
apiVersion: database.gcp.crossplane.io/v1beta1
kind: CloudSQLInstanceClass
metadata:
  name: standard-mysql
specTemplate:
  # If you change this any managed resource dynamically provisioned
  # using this class will write its connection secrets to the namespace
  # you specify. This is a required field; there is no default.
  writeConnectionSecretsToNamespace: crossplane-system
  forProvider:
    databaseVersion: MYSQL_5_7
    region: us-central1
    settings:
      tier: db-n1-standard-1
      dataDiskType: PD_SSD
      dataDiskSizeGb: 10
      ipConfiguration:
        privateNetworkRef:
          name: sample-network
  providerRef:
    name: example
```

```yaml
apiVersion: database.gcp.crossplane.io/v1beta1
kind: CloudSQLInstance
metadata:
  name: standard-mysql-negz
spec:
  writeConnectionSecretToRef:
    # If you change this the managed resource will write its connection
    # secret to the namespace you specify. This is a required field; there
    # is no default.
    namespace: crossplane-system
    name: standard-mysql-negz
  forProvider:
    databaseVersion: MYSQL_5_7
    region: us-central1
    settings:
      tier: db-n1-standard-1
      dataDiskType: PD_SSD
      dataDiskSizeGb: 10
      ipConfiguration:
        privateNetworkRef:
          name: sample-network
  providerRef:
    name: example
```

Reply:
This would require maybe making a clusterStackInstall have it's controller location be configurable instead of implied from the namespace, which has gone away now that it's going ClusterScoped.

Reply:
> This would require maybe making a clusterStackInstall have it's controller location be configurable instead of implied from the namespace

https://github.com/crossplaneio/crossplane/pull/1023/files#diff-df082ec602125906f2f5c204029cf3f2R128-R135